src/main/java/blackjack/domain/WinningDiscriminator.java

```diff
@@ -0,0 +1,70 @@
+package blackjack.domain;
+
+import blackjack.domain.gambler.Name;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import static blackjack.domain.WinningType.*;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+public class WinningDiscriminator {
+    public static final int BLACK_JACK = 21;
+    private final int dealerScore;
+    private final Map<Name, Integer> playerScores;
+
+    public WinningDiscriminator(final int dealerScore, final Map<Name, Integer> playerScores) {
+        this.dealerScore = dealerScore;
+        this.playerScores = playerScores;
+    }
+
+    public Map<Name, WinningType> judgePlayersResult() {
+        return playerScores.keySet()
+                .stream()
+                .collect(toMap(identity(), this::judgePlayerResult));
+    }
+
+    public Map<WinningType, Integer> judgeDealerResult() {
+        Map<WinningType, Integer> winningResult = createWinningResult();
+        for (final Name playerName : playerScores.keySet()) {
+            countDealerWinning(playerName, winningResult);
+        }
+        return winningResult;
+    }
+
+    private Map<WinningType, Integer> createWinningResult() {
+        return Arrays.stream(WinningType.values())
+                .collect(toMap(identity(), type -> 0));
+    }
+
+    private void countDealerWinning(final Name name, final Map<WinningType, Integer> winningResult) {
+        WinningType winningType = judgePlayerResult(name);
+        if (winningType == WIN) {
+            winningResult.put(DEFEAT, winningResult.get(DEFEAT) + 1);
+            return;
+        }
+        if (winningType == DEFEAT) {
+            winningResult.put(WIN, winningResult.get(WIN) + 1);
+            return;
+        }
+        winningResult.put(DRAW, winningResult.get(DRAW) + 1);
+    }
```

Comment:
WinningType에 reverser()와 같은 메서드를 넣어 WIN -> DEFEAT, DEFEAT -> WIN, DRAW -> DRAW로 변환해주는 메서드를 넣으면 어떨까요? 그러면 아래처럼 코드가 간단해질 것 같습니다.

```suggestion
    private void countDealerWinning(final Name name, final Map<WinningType, Integer> winningResult) {
        WinningType dealerResult = judgePlayerResult(name).reverse();
        winningResult.put(dealerResult, winningResult.get(dealerResult) + 1);
    }
```

Reply:
WinningType에 reverse를 작성하여 해당 코드를 반영하였습니다! 👍🏻